bionic/flow.py

```diff
@@ -1406,96 +1469,95 @@ def partial(*args, **kwargs):
-                check_for_bytecode_errors=False,
+                treat_bytecode_as_functional=False, check_for_bytecode_errors=False,
             )
-        elif core__versioning_mode == 'assist':
+        elif core__versioning_mode == "assist":
             return VersioningPolicy(
-                treat_bytecode_as_functional=False,
-                check_for_bytecode_errors=True,
+                treat_bytecode_as_functional=False, check_for_bytecode_errors=True,
             )
-        elif core__versioning_mode == 'auto':
+        elif core__versioning_mode == "auto":
             return VersioningPolicy(
-                treat_bytecode_as_functional=True,
-                check_for_bytecode_errors=False,
+                treat_bytecode_as_functional=True, check_for_bytecode_errors=False,
             )
         else:
-            raise ValueError(oneline(f'''
+            raise ValueError(
+                oneline(
+                    f"""
                 core__versioning_mode must be one of
                 ('manual', 'assist', 'auto');
-                got {core__versioning_mode!r}'''))
+                got {core__versioning_mode!r}"""
+                )
+            )
 
     @builder
     @decorators.immediate
     def core__persistent_cache__flow_dir(
-            core__persistent_cache__global_dir, core__flow_name):
-        return os.path.join(
-            core__persistent_cache__global_dir, core__flow_name)
+        core__persistent_cache__global_dir, core__flow_name
+    ):
+        return os.path.join(core__persistent_cache__global_dir, core__flow_name)
 
-    builder.assign('core__persistent_cache__gcs__bucket_name', None)
-    builder.assign('core__persistent_cache__gcs__enabled', False)
+    builder.assign("core__persistent_cache__gcs__bucket_name", None)
+    builder.assign("core__persistent_cache__gcs__enabled", False)
 
     @builder
     @decorators.immediate
     def core__persistent_cache__gcs__object_path():
         import getpass
-        return f'{getpass.getuser()}/bndata/'
+
+        return f"{getpass.getuser()}/bndata/"
 
     @builder
     @decorators.immediate
     def core__persistent_cache__gcs__url(
-            core__persistent_cache__gcs__bucket_name,
-            core__persistent_cache__gcs__object_path):
+        core__persistent_cache__gcs__bucket_name,
+        core__persistent_cache__gcs__object_path,
+    ):
         bucket_name = core__persistent_cache__gcs__bucket_name
         object_path_str = core__persistent_cache__gcs__object_path
 
         if bucket_name is None:
             return None
 
         path = PosixPath(bucket_name) / object_path_str
-        return f'gs://{path}'
+        return f"gs://{path}"
 
     @builder
     @decorators.immediate
-    def core__persistent_cache__local_store(
-                core__persistent_cache__flow_dir,
-            ):
+    def core__persistent_cache__local_store(core__persistent_cache__flow_dir,):
```

Comment:
super nit: comma at the end is pretty awkward now. This occurs quite a few times. Would you mind quickly replacing all `,)` with `)`?